plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java

```diff
@@ -118,8 +122,9 @@ public static Optional<String> getFilterClause(TupleDomain<ColumnHandle> tupleDo
         }
     }
 
-    private static String toPredicate(String columnName, Domain domain)
+    private static String toPredicate(PinotColumnHandle pinotColumnHandle, Domain domain, boolean forHavingClause)
     {
+        String columnName = forHavingClause ? pinotColumnHandle.getExpression() : quoteIdentifier(pinotColumnHandle.getColumnName());
```

Comment:
We should change the variable name since it might be not "column name" when `forHavingClause` is true.